plugin-server/src/config/config.ts

```diff
@@ -38,7 +44,9 @@ export function getDefaultConfig(): PluginsServerConfig {
         KAFKA_SASL_MECHANISM: null,
         KAFKA_SASL_USER: null,
         KAFKA_SASL_PASSWORD: null,
-        KAFKA_CONSUMPTION_TOPIC: KAFKA_EVENTS_PLUGIN_INGESTION,
+        KAFKA_CONSUMPTION_TOPIC: isOverflowMode()
```

Comment:
In development we will only have one "deployment". With this setup I think we'll only be consuming from one of the topics.

Can we instead add an additional consumer? something like if('ingestion-overflow' in capabilities) then start an additional consumer?

Reply:
I'm having trouble doing this. I believe this is where all consumers are started: https://github.com/PostHog/posthog/blob/master/plugin-server/src/main/pluginsServer.ts#L238, and I think I should copy the one we start for event ingestion and change the topic. But I can't understand which function is starting this consumer.

Reply:
Right! I think the logic here is buried a little in [this call](https://github.com/PostHog/posthog/blob/master/plugin-server/src/main/pluginsServer.ts#L286)

I feel it would be less confusing to have the capabilities differentiation happening within the top level i.e. in `pluginsServer.ts`. That's not how that part of the code is working now, so either a refactor, or just going with how the capabilities are handled there at the moment would do.

Reply:
Right, I'm quite confused about this part of the code. Is this only supposed to be for dev? Will we have two deployments in prod?

Reply:
We'll have two consumers, both in dev and production. But in production we'll run the consumers on separate deployments based on the value of `PLUGIN_SERVER_MODE`.

Reply:
So we only create two consumers in dev, as in prod each consumer will be created by each separate deployment.

Reply:
Otherwise we would end up with 4 consumers in prod.

Reply:
I have no idea how to achieve this :sweat_smile:

Reply:
I mean, I'll figure it out eventually, it's just going to take a while. So I would appreciate some help to ship this relatively soon.

Reply:
hah in the current capabilities based setup I would:

 1. add a new "capability" to https://github.com/PostHog/posthog/blob/master/plugin-server/src/capabilities.ts#L8 for the null case e.g. dev
 2. add a new "mode" for ingestion-overflow, with the new overflow capability
 3. add a conditional on the topic that is being used [here](https://github.com/PostHog/posthog/blob/master/plugin-server/src/main/ingestion-queues/kafka-queue.ts#L49) (or refactor to be simpler with this branching happening within `pluginServer.ts`)

Reply:
fwiw I've added an overview of how this will be handled on rollout [here](https://github.com/PostHog/charts-clickhouse/pull/682/files#diff-875061ff1914ed677171e0961973370a1b63718e285facb655f2b2bd458d0b6cR12)

It's a bit clunky so I you have some ideas... otherwise I _think_ it will do.

Reply:
I'll give this a shot, thanks for the pointers!